sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -584,7 +632,29 @@ class ResolveSessionCatalog(
   object SessionCatalogAndTable {
     def unapply(nameParts: Seq[String]): Option[(CatalogPlugin, Seq[String])] = nameParts match {
       case SessionCatalogAndIdentifier(catalog, ident) =>
-        Some(catalog -> ident.asMultipartIdentifier)
+        if (nameParts.length == 1) {
+          // If there is only one name part, it means the current catalog is the session catalog.
+          // Here we return the original name part, to keep the error message unchanged for
+          // v1 commands.
```

Comment:
I can remove this, but then I need to update many tests slightly to update the error message. We can leave it to 3.1.

Reply:
We can remove https://github.com/apache/spark/pull/27550/files#diff-2e07be4d73605cb1941153441a0c0c14R572 and https://github.com/apache/spark/pull/27550/files#diff-2e07be4d73605cb1941153441a0c0c14R651 if we return qualified name here.